lib/internal/modules/esm/fetch_module.js

```diff
@@ -148,7 +148,7 @@ function fetchWithRedirects(parsed) {
         err.message = `Cannot find module '${parsed.href}', HTTP 404`;
         throw err;
       }
-      if (res.statusCode < 200 || res.statusCode >= 400) {
+      if (res.statusCode < 200 || res.statusCode >= 300) {
```

Comment:
I think an explanation for this `if` condition would be helpful. Is this the correct reason?

```suggestion
      // The 3xx codes that we support were handled above in isRedirect
      // This condition catches all unsupported status codes
      if (res.statusCode < 200 || res.statusCode >= 300) {
```

Reply:
Added a comment via https://github.com/nodejs/node/pull/43742/commits/640b85da99c99b66e7bff55390a85dffc3b9f0f8.
I think, this condition can be read properly and unambiguously as "we need 2xx code", so there is no difference between `1xx`, `3xx`, `5xx` or any invalid number.
What we might need to point out is that isRedirect check above handled only machine-readable redirects with `Location`, so we want to filter them again. If this part will be refactored into some `validateStatusCode` that throws meaningful errors for different coderanges, we don't want these to be omitted.